### PR TITLE
[docs] 2018 refresh

### PR DESCRIPTION
**Pull request by jacknagz**
to: @ryandeivert @chunyong-lin @mime-frame 
cc: @airbnb/streamalert-maintainers
size: medium

## Background

Go through the documentation and remove technical debt, fix grammar, and modify style.

## Testing

Locally via the `test_the_docs.sh` script

### CODE REVIEW

File: docs/source/account.rst [hunk 1]
```diff
@@ -4,87 +4,64 @@ Account
...
 
-Configuration
--------------
-As outlined above, choose or create the AWS account you will use to house the StreamAlert infrastructure.
-
-If you are interested in simply demo'ing StreamAlert, you can create a hassle free-tier AWS account `here <https://aws.amazon.com/free/>`_.
-
-account_id
-~~~~~~~~~~
+If do not currently have an AWS account and you are interested in simply testing StreamAlert, you can create a hassle free-tier AWS account `here <https://aws.amazon.com/free/>`_.
```
Comment (chunyong-lin) [inline docs/source/account.rst]: `If you do not`

File: docs/source/account.rst [hunk 2]
```diff
@@ -4,87 +4,64 @@ Account
 Background
 ----------
 
-StreamAlert can accept ingest, analyze and alert on data from a wide range of environments and operating systems.
+StreamAlert can ingest, analyze and alert on data from a wide variety of environments, applications, or systems.
```
Comment (mime-frame) [inline docs/source/account.rst]: or => and

File: docs/source/account.rst (hunk 1 above)
Comment (mime-frame) [inline docs/source/account.rst]: If you want to try out StreamAlert in a test environment, you can use the AWS Free tier to get some hands-on experience.

Comment (ryandeivert) [inline docs/source/account.rst]: I think maybe this should be `hassle-free free tier` account but then that sounds kinda silly

File: docs/source/clusters.rst [hunk 3]
```diff
@@ -4,58 +4,55 @@ Clusters
...
-Option \(2\) is encouraged because it provides segmentation for ingestion, analysis and storage, on a per-cluster basis, ensuring that folks only have access to the infrastructure and data they need to get their job done.
+Option \(2\) is encouraged because it provides segmentation for ingestion and analysis, which allows for the best possible performance during processing.
 
 Configuration
 -------------
 
-Each cluster lives in its own ``json`` file in the ``conf/clusters`` directory.
-
-A cluster file contains name, region, modules, and outputs.
+Each cluster file lives in its own ``JSON`` file in the ``conf/clusters`` directory, and contains a name, region, modules, and outputs.
```
Comment (ryandeivert) [inline docs/source/clusters.rst]: `outputs` in this sentence is a little misleading here - can we omit or clarify what kind of `outputs` this is referring to?

**Comment by jacknagz** [comment]
Thanks for reviewing @chunyong-lin @mime-frame @ryandeivert !!!! 🐑

**Pull request merged**